edx_when/api.py

```diff
@@ -296,14 +364,17 @@ def get_overrides_for_user(course_id, user):
         yield {'location': udate.content_date.location, 'actual_date': udate.actual_date}
 
 
-def set_date_for_block(course_id, block_id, field, date_or_timedelta, user=None, reason='', actor=None):
+def set_date_for_block(
```

Comment:
Should this method also be clearing the cache when a new date is set? I know we do it when we clear dates, but wouldn't this be a similar process?

Reply:
The assumption I'd made here is that a `set_date_for_block` would be preceded by a course publish - and therefore a new `published_version` would be passed-in. This new `published_version` would leave the old cached value behind. Clearing all course dates on blocks might *not* necessarily be preceded by a course publish.